lib/web/fetch/index.js

```diff
@@ -133,116 +177,128 @@ function fetch (input, init = undefined) {
   let requestObject
 
   try {
-    requestObject = new Request(input, init)
-  } catch (e) {
-    p.reject(e)
-    return p.promise
-  }
+    publishIfContext('start', context)
```

Comment:
again: Why do you have to put everything in a giant try catch block?

Reply:
It's necessary to ensure the `end` event always occurs, even if the function fails for some reason.